src/App.tsx

```diff
@@ -1,60 +1,156 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+// import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+import { TodoList } from './components/TodoList';
+
+export function usersInTodos(): Todo[] {
+  return todosFromServer.map(todo => ({
+    ...todo,
+    user: usersFromServer.find(user => user.id === todo.userId),
+  }));
+}
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(usersInTodos());
+  const [title, setTitle] = useState('');
+  const [user, setUser] = useState(0);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasChangeError, setHasChangeError] = useState(false);
+
+  const resetInputSFields = () => {
+    setTitle('');
+    setUser(0);
+  };
+
+  const checkEmptyFileds = () => {
+    if (!user || !title.trim()) {
+      if (!user) {
+        setHasChangeError(true);
+      }
+
+      if (!title.trim()) {
+        setHasTitleError(true);
+      }
+    }
+
+    return user && title.trim();
+  };
+
+  const findTheBiggestId = () => {
```

Comment:
it's better to name it like "findTheNextId" because you find biggest and then add 1))